server/src/test/java/org/elasticsearch/common/xcontent/BaseXContentTestCase.java

```diff
@@ -843,8 +843,8 @@ public void testEnsureNotNull() {
     }
 
     public void testEnsureNoSelfReferences() throws IOException {
-        CollectionUtils.ensureNoSelfReferences(emptyMap());
-        CollectionUtils.ensureNoSelfReferences(null);
+        builder().map(emptyMap());
```

Comment:
I fixed these to call the functions that are supposed to be under test here, but I did not go so far as to introduce a message hint to the XContentBuilder version of `ensureNoSelfReferences()`, just because that's where I arbitrarily drew the line on scope. Happy to revisit if you'd like.